lib/cfn-nag/custom_rules/IamManagedPolicyPassRoleWildcardResourceRule.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class IamManagedPolicyPassRoleWildcardResourceRule < BaseRule
+  def rule_text
+    'IAM managed policy should not allow a * resource with PassRole action'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F40'
+  end
+
+  def passrole_action?(statement)
+    statement.actions.find { |action| action == 'iam:PassRole' }
```

Comment:
i'm not sure we've ever written a rule before that spells out particular actions?  i'm thinking we need to "relax" this equality check s.t. it matches up against a partial wildcard.  for example, iam:Pass*, iam:PassR* should match.  there's probably something generic we can write, but for now, a naive regex match is good enough?

Reply:
Created a iam_permutations utility.